iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/ReferenceTerminals.java

```diff
@@ -39,7 +40,7 @@ default String getName() {
     ReferenceTerminals addReferenceTerminal(Terminal terminal);
 
     /**
-     * Deletes all defined reference terminals in the network for the current variant
+     * Deletes all defined reference terminals in the network and all its subnetworks for the current variant
```

Comment:
this is what e.g. OLF will want do

Reply:
It depends on which network the load flow is run. If you run it on a merged network, you can/have to leave the reference terminals of the subnetworks.

Reply:
static void reset(Network network) {
    network.getVoltageLevels().forEach((vl) -> {
        reset(vl, (Terminal)null);
    });
}

static void reset(VoltageLevel voltageLevel, Terminal terminal) {
    Objects.requireNonNull(voltageLevel);
    SlackTerminal st = (SlackTerminal)voltageLevel.getExtension(SlackTerminal.class);
    if (st == null && terminal != null) {
        ((SlackTerminalAdder)voltageLevel.newExtension(SlackTerminalAdder.class)).withTerminal(terminal).add();
    } else if (st != null) {
        st.setTerminal(terminal, true);
    }

}  
-> we need also to align here if we clean the subnetworks too.

Reply:
done / SlackTerminal.reset modified to reset also the subnetworks.
also updated PR title & description